crates/solidity/testing/sanctuary/src/tests.rs

```diff
@@ -81,7 +81,8 @@ pub fn run_test(file: &SourceFile, events: &Events) -> Result<()> {
     // Heuristic: ignore wrongly scraped sanctuary files that
     // contain HTML with a Cloudflare email obfuscation attribute:
     // https://github.com/tintinweb/smart-contract-sanctuary/issues/32
-    if source.contains("data-cfemail=") {
+    // Also ignore wrongly obfuscated `\r@` that leads to an invalid escape sequence:
+    if source.contains("data-cfemail=") || source.contains("\\[email protected]") {
```

Comment:
We already replace `email protected` below, so that we can still test the contract. Am I missing something?

Reply:
yes, notice the `\\`: the comment mentions explicitly that obfuscating `\r@` that leads to an invalid escape: we end up with `\[...` and `\[` is invalid in a string.

I didn't put it after the `[email protected]` -> `[email-protected]` because we'd introduce yet another "special case" and this if here is about scraping that introduces parse errors (e.g. `\[`), so exactly this case.

Reply:
I understand why we are skipping `\[`, but not sure how that results from obfuscating `\r@`?

In any case, not blocking. Happy to follow up on this later.

Reply:
I tried to improve the wording in https://github.com/NomicFoundation/slang/pull/946/commits/eeac4b9213a5cc859896ac44e90254e7d8767bcb to make it more obvious why the original valid sequence ends up malformed by obfuscating `r@`, specifically, in this case.